03-algorithms-sorting/insertion_sort.rb

```diff
@@ -0,0 +1,20 @@
+class InsertionSort
+  def insertion_sort(array)
+    sorted = [array.delete_at(0)]
+
+    for val in array
+      index = 0
+      while index < sorted.length  # loops through 'sorted' array
+        if val <= sorted[index]  # if value is less than item in 'sorted'
+          sorted.insert(index, val)  # insert the value into 'sorted'
```

Comment:
Also, that is based on the Bloc's code. And this is what I coded from scratch. I didn't use the `.insert` method here because I didn't know that kind of built-in method already exists!

```ruby
def insert_sort(collection)
  sorted = [collection[0]]
  for i in 1...collection.length
    if collection[i] > sorted[-1]
      sorted.push(collection[i])
    elsif collection[i] <= sorted[0]
      sorted.unshift(collection[i])
    else
      for j in 1...sorted.length
        if collection[i] <= sorted[j]
          sorted = sorted[0..j-1] + [collection[i]] + sorted[j..-1]
          break
        end
      end
    end
  end
  sorted
end
```